src/python/pants/base/deprecated.py

```diff
@@ -100,7 +100,7 @@ def _get_frame_info(stacklevel, context=1):
 
 
 def warn_or_error(removal_version, deprecated_entity_description, hint=None, stacklevel=3,
-                  frame_info=None, context=1, ensure_stderr=False):
+                  frame_info=None, ensure_stderr=False):
```

Comment:
The context arg is used by prod code.

Reply:
I was able to figure out how to resolve this by monkey patching `warnings.showwarning` in 47b0044. `warn_explicit()` just doesn't provide the `line` argument -- so I was able to retain the current behavior while now respecting warnings filters with https://github.com/python/cpython/blob/fd83a823a6f268dc97ee2bf7d8a1a88d948446e5/Lib/warnings.py#L300.